File: js/stockPicker.js
exports.picker = function(prices) {
    let sellIndex = 0;
    let buyIndex = 0;
    let highestProfit = 0;
    let LowestPriceIndx = 0;
    // start at the next index so you can catch all cases
    for (let i = 1; i < prices.length; i++) {
        const currentPrice = prices[i];
        if (currentPrice < prices[LowestPriceIndx]) {
            LowestPriceIndx = i;
        } else {
            let profit = currentPrice - prices[LowestPriceIndx];
            if (profit > highestProfit) {
                highestProfit = profit;
                sellIndex = i;
                buyIndex = LowestPriceIndx;
            }
        }
    }
    return [buyIndex, sellIndex];
}

File: python/stock_picker.py
def picker(prices):
    sell_index = 0
    buy_index = 0
    max_profit = 0
    lowest_index = 0
    for i in range(1, len(prices)):
        current_price = prices[i]
        if current_price < prices[lowest_index]:
            lowest_index = i
        else:
            profit = current_price - prices[lowest_index]
            if profit > max_profit:
                max_profit = profit
                sell_index = i
                buy_index = lowest_index
    return [buy_index, sell_index]
